sdk/iot/provisioning/samples/src/paho_iot_provisioning_example.c

```diff
@@ -262,6 +220,119 @@ static int register_device()
+      // otherwise topicLen is 0.
+      topicLen = (int)strlen(topicName);
+    }
+
+    printf("Message arrived\n");
+    printf("     topic (%d): %s\n", topicLen, topicName);
+    printf("   message (%d): ", message->payloadlen);
+    print_buffer(az_span_init(message->payload, message->payloadlen));
+
+    az_iot_provisioning_client_register_response response;
+    char query_topic[256];
+
+    if (az_failed(
+            rc = az_iot_provisioning_client_parse_received_topic_and_payload(
+                &provisioning_client,
+                az_span_init((uint8_t*)topicName, topicLen),
+                az_span_init((uint8_t*)message->payload, message->payloadlen),
+                &response)))
+    {
+      printf("Message from unknown topic, return code %d\n", rc);
+      return rc;
+    }
+
+    if (az_span_is_content_equal(response.registration_state, AZ_SPAN_FROM_STR("assigning")))
+    {
+      if (az_failed(
+              rc = az_iot_provisioning_client_query_status_get_publish_topic(
+                  &provisioning_client, &response, query_topic, sizeof(query_topic), NULL)))
+      {
+        printf("Failed to get operation status topic, return code %d\n", rc);
+        return rc;
+      }
+
+      printf("Querying after %u seconds...\n", response.retry_after_seconds);
+      sleep_seconds(response.retry_after_seconds);
+
+      if ((rc = MQTTClient_publish(mqtt_client, query_topic, 0, NULL, 0, 0, NULL))
+          != MQTTCLIENT_SUCCESS)
+      {
+        printf("Failed to publish get_operation_status, return code %d\n", rc);
+        return rc;
+      }
+    }
+    else if (az_span_is_content_equal(response.registration_state, AZ_SPAN_FROM_STR("assigned")))
+    {
+      printf("SUCCESS - Device provisioned:\n");
+      printf("\tHub Hostname: ");
+      print_buffer(response.registration_information.assigned_hub_hostname);
+      printf("\tDevice Id: ");
+      print_buffer(response.registration_information.device_id);
+      registration_complete = true;
+    }
+    else // unassigned, failed or disabled states
+    {
+      printf("ERROR - Device Provisioning failed:\n");
+      printf("\tRegistration state: ");
+      print_buffer(response.registration_state);
+      printf("\tLast operation status: %d\n", response.status);
+      printf("\tOperation ID: ");
+      print_buffer(response.operation_id);
+      printf("\tError code: %u\n", response.registration_information.extended_error_code);
+      printf("\tError message: ");
+      print_buffer(response.registration_information.error_message);
+      printf("\tError timestamp: ");
+      print_buffer(response.registration_information.error_timestamp);
+      printf("\tError tracking ID: ");
+      print_buffer(response.registration_information.error_tracking_id);
+      if (response.retry_after_seconds > 0)
+      {
+        printf("\tRetry-after: %u seconds.", response.retry_after_seconds);
+      }
+
+      registration_complete = true;
```

Comment:
is this false? if not, please add comments

Reply:
It's not: the registration operation is complete. It can complete with success or error.

Reply:
I'll change the var name to is_operation_complete since it's confusing.